api/src/main/java/org/apache/iceberg/transforms/Transforms.java

```diff
@@ -192,4 +230,10 @@ private Transforms() {
   public static <T> Transform<T, Void> alwaysNull() {
     return VoidTransform.get();
   }
+
+  private static void checkZoneOffsetIsUTC(ZoneOffset zoneOffset) {
+    Preconditions.checkArgument(zoneOffset == null || zoneOffset.getTotalSeconds() == 0,
+        "Expect zone offset is null or UTC, but is", zoneOffset);
```

Comment:
The offset should always be null because it is invalid to call `day` with a date field and a zone offset. Even if the offset is UTC, it is not allowed.